### PR TITLE
[CI] Add e2e tests to Settings > Service Meshes / consul

### PR DESCRIPTION
**Pull request by ramrodo**
Signed-off-by: Rodolfo Martínez Vega <rodomar@outlook.com>

**Description**

This PR adds e2e tests to **Settings > Service Meshes** to configure the `meshery-consul` adapter 

This PR is related to #1037

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
I wonder if we might be able to have an authentication flow included to invoke the Meshery remote provider. There are service/robot accounts we can use to authenticate, if so.

**Comment by ramrodo** [comment]
> I wonder if we might be able to have an authentication flow included to invoke the Meshery remote provider. There are service/robot accounts we can use to authenticate, if so.

Sure, sounds good. Where can I see those service/robot accounts? Also, is there a previous example to use them?

**Comment by leecalcote** [comment]
@ramrodo there is a little bit of cleanup needed in this doc, but it'll be a core resource to work with - [Meshery Build & Release Strategy](https://docs.google.com/document/d/11nAxYtz2SUusCYZ0JeNRrOLIxkgmmbUVWz63MBZV2oE/edit).

**Comment by leecalcote** [comment]
The [Meshery Cloud Build & Release Strategy](https://docs.google.com/document/d/1sq05a0YQco33yr96WYpw9x5Ra9zvyZfK5GUMxdyK12E/edit#) is more up-to-date. It speaks only to the "Meshery" Remote Provider, though.

**Comment by ramrodo** [comment]
@leecalcote Finally, the tests passed! Thanks for the **Cypress Dashboard**, it was very helpful to debug the errors.

And talking about your comment:

> I wonder if we might be able to have an authentication flow included to invoke the Meshery remote provider. There are service/robot accounts we can use to authenticate, if so.

I think we can implement an authentication flow using service/robot account but in another PR. This changes only set up the meshery adapter for **consul** in a KinD cluster.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -202,6 +214,11 @@ jobs:
           start: npm start
           wait-on: http://localhost:3000
           spec: cypress/integration/e2e/**/*
+          record: true
+          tag: node-${{ matrix.node }}
+        env:
+          CYPRESS_RECORD_KEY: "a25e3b24-8ff8-41e6-be40-5e33e33596e2"
```
Comment (leecalcote) [inline .github/workflows/ci.yml:220]: @ramrodo is this a key that we might store in a GitHub secret?

Comment (ramrodo) [reply .github/workflows/ci.yml:220]: That is a good idea @leecalcote! I don't have permissions on Github to do that yet. Could you help me with that?

Comment (leecalcote) [reply .github/workflows/ci.yml:220]:
```suggestion
          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
```

Perfect. I have added this as an organization-level secret. 👍

**Comment by ramrodo** [comment]
@leecalcote Do you have something else to add? If not, I think we can merge this PR :)

**Review by leecalcote — APPROVED** [review]
This is fantastic. :)

**Pull request merged**